### PR TITLE
Jason CanCanCan Permissions

### PR DESCRIPTION
**Pull request by JasonBaoZ**
## Feature Name
Fixed the routes so that they're RESTful. Changed some minor things (e.g. artist edit profile was in an update route when it should be in an edit route). Think I changed some other routes that don't make sense (e.g. home -> index). Pretty sure the permissions work (if you try to edit a profile that isn't yours it will redirect to the buyer sign_in screen). 

### Related PRs
None

### Migrations
None

### Tests Performed, Edge Cases
Tried to edit works and profile that wasn't my own. Was redirrected.

### Screenshots

CC: @clairelin135

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,38 +5,39 @@
...
 
-  get '/buyers/:id' => 'buyers#show', as: :buyerid
+  resources :requests, only: [:index] do
+    collection do
+      get 'types', action: :get_type_enum
+    end
+  end
 
-  get '/commissions/types' => 'commissions#get_type_enum'
-  get '/commissions' => 'artists#commissions', as: :commissions
```
Comment (clairelin135) [inline config/routes.rb]: I think that this route no longer works
